ouroboros-network/src/Ouroboros/Network/PeerSelection/Governor/Monitor.hs

```diff
@@ -617,7 +625,10 @@ monitorLedgerStateJudgement PeerSelectionActions{ readLedgerStateJudgement }
       return $ \_now ->
         Decision {
           decisionTrace = [TraceLedgerStateJudgementChanged lsj],
-          decisionJobs  = [],
+          decisionJobs = case (lsj, ledgerPeerSnapshot) of
+                           (TooOld, Just ledgerPeerSnapshot') ->
+                             [jobVerifyPeerSnapshot ledgerPeerSnapshot' ledgerCtx]
+                           _otherwise -> [],
```

Comment:
This job is fired when `ledgerStateJudgement` changes, it doesn't mean that's when the snapshot was taken.  It's quite possible that the node was started with a valid snapshot taken 3 months behind, which is well past the `ledgerStateJudgement` change.

We have to have to wait for the exact slot in which the ledger snapshot was taken.  Also note that we need to obtain ledger peers in the same `STM` transaction, otherwise if this is done in a separate thread it could happen that we will not be comparing the snapshot with the right ledger.

Reply:
The job which is started doesn't verify immediately, it does wait until we are >= the slot where the snapshot was taken, and then it does the verification. It does obtain the ledger peers in the same transaction as well, so all your concerns are indeed addressed.